pkg/service/configuration_service_routine.go

```diff
@@ -1,21 +1,14 @@
 package service
 
 import (
-	"errors"
 	"fmt"
 
 	"github.com/aerospike/backup/pkg/model"
 )
 
 // AddRoutine
 // adds a new BackupRoutine to the configuration if a routine with the same name doesn't already exist.
-func AddRoutine(config *model.Config, newRoutine *model.BackupRoutine) error {
-	if newRoutine.Storage == "" {
-		return errors.New("storage is empty")
-	}
-	if newRoutine.SourceCluster == "" {
-		return errors.New("cluster is empty")
-	}
+func AddRoutine(config *model.Config, name *string, newRoutine *model.BackupRoutine) error {
```

Comment:
You always dereferene the name argument, so it's better to take it by copy as a parameter (in all places).